lib/winston-papertrail.js

```diff
@@ -136,6 +136,40 @@ var Papertrail = exports.Papertrail = function (options) {
     // Open the connection
     connectStream();
 
+    function cleanup() {
+        self._erroring = true;
+        try {
+            if (self.socket) {
+                self.socket.destroy();
+
+                //this throws some uncatchable thing, eww.
+                //self.socket.end();
+                self.socket = null;
+            }
+        }
+        catch (e) { }
+
+        try {
+            if (self.stream) {
+                //we're cleaning up, don't loop.
+                self.stream.removeListener('end', connectStream);
+                self.stream.removeListener('error', onErrored);
+
+                self.stream.destroy();
+                self.stream = null;
+            }
+        }
+        catch (e) { }
```

Comment:
:)

Maybe it's better to discuss what stream or socket.destroy will throw and then we can be more accurate about the value of throwing here.